source/extensions/quic_listeners/quiche/platform/quic_logging_impl.cc

```diff
@@ -49,6 +58,12 @@ void SetVerbosityLogThreshold(int new_verbosity) {
   g_verbosity_threshold.store(new_verbosity, std::memory_order_relaxed);
 }
 
+bool IsDFatalExitDisabled() { return g_dfatal_exit_disabled.load(std::memory_order_relaxed); }
```

Comment:
Why do we use relaxed ordering here? I'm pretty skeptical about the use of atomics unless there is some very clearly documented (and hopefully enforced) use guidelines around when things can be set/get relative to other code. Memory ordering models can do surprising things.

Reply:
This should be fine since we don't intent to use the value of g_dfatal_exit_disabled to synchronize multiple threads. I'm just following how google3 does it: http://shortn/_2FoTWxZWDc

Reply:
@wu-bin how does the DFATAL condition propagate from Envoy -> QUICHE? I would like to understand if we could ever have a mismatch in behavior around DFATAL based on build types we have setup for Envoy and what QUICHE is calculating.

Reply:
DFATAL should only be used in QUICHE code, it is not supposed to be used directly in Envoy.  g_dfatal_exit_disabled should only be used by unit tests who wants to catch the error message of a DFATAL log(instead of aborting the program). g_dfatal_exit_disabled should always be false in non-unit-test environments.

Reply:
That wasn't quite my question; I'm wondering if the QUICHE code might fatal out in non-debug modes for Envoy's build, etc. How does QUICHE decide how to condition on DFATAL?

Reply:
What I'm still wondering about is "how does QUICHE detect that it is in debug mode when built under Bazel with Envoy?". Do you have any code pointers to that?

Reply:
It uses "#ifdef NDEBUG" for that. For example, in line 44 of this file, you can see "g_dfatal_exit_disabled" is not used in release mode.

Reply:
Great, makes sense.